andela/ah-technocrats
# #161966607 reset password

### What does this pull request do?
 Add the password reset feature
### Description of the tasks to be completed?
- add functionality to enable users to reset their password
- send an email with a link to reset the password
 - add tests to test the above functionalities.
### How should this be manually tested?
- Clone the repo `git clone https://github.com/andela/ah-technocrats.git`
- `cd ah-technocrats`
- Checkout out to this branch `git checkout ft-password-reset-161966607 `
- Make virtual environment `python3 -m venv venv`
- Activate the virtual environment `source venv/bin/activate`
- Install all the dependencies from the requirements file ` pip install -r requirements.txt`
- Make migrations  `python manage.py makemigrations`
- Apply the migrations `python manage.py migrate`
- Run the server  `python manage.py runserver`

- Using Postman, register a user with a valid email address.
<img width="768" alt="screen1" src="https://user-images.githubusercontent.com/25079238/49515643-8b4be080-f8a8-11e8-8c2b-57ff568c0582.png">

- To make a password reset make a post request with your email address to the route
  `POST / api/users/forgot_password/`

<img width="631" alt="screen2" src="https://user-images.githubusercontent.com/25079238/49516019-905d5f80-f8a9-11e8-80a0-8bba8731614d.png">

- You should recieve a link in your inbox if the email you provided is valid.
<img width="892" alt="screen3" src="https://user-images.githubusercontent.com/25079238/49516240-327d4780-f8aa-11e8-81e7-0bcb1f9c2e57.png">

- Following the link should lead you to a Django Rest Framework  Change Password Page where you can change your password. The new password should have at least 8 characters, one number, one letter, and one special character.
<img width="1111" alt="screen4" src="https://user-images.githubusercontent.com/25079238/49516500-d1a23f00-f8aa-11e8-8074-53d5f92b3e48.png">

- You can also copy the link to Postman and make a PUT request with the new password.

### Any background context you want to provide?
- The user requires a method to reset a password in case they forget it.
- The reset feature needs to be secure and only accessible to the user via their email.
### What are the relevant Pivotal Tracker Stories
[#161966607](https://www.pivotaltracker.com/story/show/161966607)
# Checklist:
